weaver/main.go

```diff
@@ -98,5 +98,17 @@ func main() {
 		}()
 	}
 
-	log.Fatal(router.Run(conf.HTTPAddr))
+	server := &http.Server{
+		Addr:    conf.HTTPAddr,
+		Handler: router,
+	}
+
+	go log.Fatal(server.ListenAndServe())
+
+	sigChan := make(chan os.Signal)
+	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+
+	<-sigChan
+	fmt.Println("Received sigterm, gracefully shutting down")
+	server.Shutdown(context.Background())
```

Comment:
+1 good code

Reply:
oh really?

Reply:
yes, very graceful

Reply:
turns out it isn't graceful at all and go sucks